docs/DeveloperGuide.md

```diff
@@ -4,30 +4,274 @@
+
+<img src = "images/ArchitectureSequenceDiagram.png">
+
+The above **Sequence Diagram** shows how the different components of the system interact with one
+another in the scenario when the command `add -e event -t 1200 -v venue` is executed.
+
+### UI component
+
+<img src = "images/UiClassDiagram.png">
+
+The `UI` component comprises an Ui class, as shown in the above **Class Diagram**.
+
+The `UI` does the following:
+
+* Take in command input from the user and pass the input to `Main`.
+* Show output messages from `Command` to the user after command execution.
+
+### Storage component
+
+<img src = "images/StorageClassDiagram.png">
+
+The `Storage` component comprises a Storage class, as shown in above **Class Diagram**.
+
+The `Storage` does the following:
+
+* Load events and participants information from a text file and save it to `EventList` list.
+* Save events from `EventList` list to a text file
+
+### Command component
+
+The `Command` component and its component classes are shown in the below **Class Diagram**:
+
+<img src = "images/CommandClassDiagram.png">
+
+The `Command` component does the following:
+
+* Handle the execution of the user command through interactions with `Event`.
+* Provides a command output message to `Ui` after the command execution.
+
+The interactions between `Command` and other commands in the system is shown in the following _Sequence Diagram_:
+
+<img src = "images/CommandSequenceDiagram.png">
+
+### Event component
+
+The `Event` component and its component classes are shown in the below **Class Diagram**:
+
+<img src = "images/EventClassDiagram.png">
+
+The `Event` component does the following:
+* Handle the addition, removal and marking of events stored in `EventList`.
+* Add, remove and mark participants for a specific `Event` in `EventList`.
+
+## Implementation
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### List feature[TBD]
+
+The `list` feature allows users to view all scheduled events in the system. 
+It is implemented in the `ListCommand` class, which extends the base `Command` class and formats the output to display all events. 
+Internally, the `list` operation does not modify any data but simply retrieves and displays the information from `EventList`.
+
+The `ListCommand` class performs the following key operations:
+
+* ListCommand#execute() — Generates a formatted message displaying all events in the list.
+
+These operations are accessible through the `Command` and can be invoked when the list command is entered by the user.
+
+#### Feature Implementation
+
+Given below is an example usage scenario and the behavior of the list feature at each step:
+
+1. User Command Input:
+The user enters the command `list` to view all scheduled events.
+
+2. Command Recognition:
+The `COMMAND_WORD` is set to "list", enabling the system to recognize the command input and invoke `ListCommand`.
+
+3. Execution of ListCommand#execute():
```

Comment:
Should method calls be formatted as a code block (i.e. enclose with ``) as well?